File: 1134-ArmstrongNumber/1134-ArmstrongNumber.java
/*

1134. Armstrong Number
Easy

The k-digit number N is an Armstrong number if and only if the k-th power of each digit sums to N.

Given a positive integer N, return true if and only if it is an Armstrong number.

Example 1:

Input: 153
Output: true
Explanation: 
153 is a 3-digit number, and 153 = 1^3 + 5^3 + 3^3.

Example 2:

Input: 123
Output: false
Explanation: 
123 is a 3-digit number, and 123 != 1^3 + 2^3 + 3^3 = 36.

Note:

    1 <= N <= 10^8

*/

class Solution {
    public boolean isArmstrong(int N) {
        int num = N, sum = 0;
        int len = String.valueOf(N).length();
        while(num != 0){
            sum += Math.pow(num%10, len);
            num = num / 10;
        }
        return sum == N;
    }
}

File: 1134-ArmstrongNumber/1134-ArmstrongNumber.py
"""

1134. Armstrong Number

Given an integer n, return true if and only if it is an Armstrong number.

The k-digit number n is an Armstrong number if and only if the kth power of each digit sums to n.

Example 1:

Input: n = 153
Output: true
Explanation: 153 is a 3-digit number, and 153 = 13 + 53 + 33.

Example 2:

Input: n = 123
Output: false
Explanation: 123 is a 3-digit number, and 123 != 13 + 23 + 33 = 36.

Constraints:

    1 <= n <= 108

"""

class Solution:
    def isArmstrong(self, n: int) -> bool:
        num = 0
        for ch in str(n):
            num += int(ch) ** len(str(n))
        return num == n
